src/nock/matcher.ts

```diff
@@ -427,16 +432,31 @@ export async function matchTraffic(
 
   if (match.valid) {
     // do not save new recording as there were no breaking changes found
+    // only write recordings with decoded responses if they do not exist
+    if (
+      parseBooleanEnv(process.env.DECODE_RESPONSE) &&
+      !(await exists(oldRecordingDecodedPath))
+    ) {
+      await writeDecodedRecordings(oldRecordingPath, newTraffic);
```

Comment:
Issue: Shouldn't this be
```suggestion
      await writeDecodedRecordings(oldRecordingDecodedPath, newTraffic);
```

Reply:
This is also resolved inside that function.